src/Http/Http.Extensions/test/RequestDelegateFactoryTests.cs

```diff
@@ -2889,6 +2889,49 @@ public async Task RequestDelegateWithBindAndImplicitBodyRejectsNonJsonContent(bo
             }
         }
 
+        public static IEnumerable<object?[]> DateTimeDelegates
+        {
+            get
+            {
+                string dateTimeParsing(DateTime time) => $"Time: {time.ToString("O")}, Kind: {time.Kind}";
+                string dateTimeOffsetParsing(DateTimeOffset time) => $"Time: {time.ToString("O")}";
+                string dateOnlyParsing(DateOnly time) => $"Time: {time.ToString("O")}";
+                string timeOnlyParsing(TimeOnly time) => $"Time: {time.ToString("O")}";
+
+                return new List<object?[]>
+                {
+                    new object?[] { (Func<DateTime, string>)dateTimeParsing, "09/20/2021 16:34:09", "Time: 2021-09-20T16:34:09.0000000Z, Kind: Utc" },
+                    new object?[] { (Func<DateTimeOffset, string>)dateTimeOffsetParsing, "09/20/2021 16:34:09 +00:00", "Time: 2021-09-20T16:34:09.0000000+00:00" },
+                    new object?[] { (Func<DateOnly, string>)dateOnlyParsing, "9/20/2021", "Time: 2021-09-20" },
+                    new object?[] { (Func<TimeOnly, string>)timeOnlyParsing, "4:34 PM", "Time: 16:34:00.0000000" },
+                };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(DateTimeDelegates))]
```

Comment:
I'm generally not a fan of using Theory case for testing what amounts to different code paths here. Could we expand these and write bespoke tests? That way if one of these changes in the future you don't have to figure out how to filter out an exact test case using `dotnet test`. You can always a helper function to re-use code.

Reply:
Yep -- this is fair. I think it is sufficient to split it up into a set of tests for `DateTime`/`DateTimeOffset` and another set of tests for `DateOnly`/`TimeOnly`.